envs/monkey_zoo/blackbox/test_configurations/utils.py

```diff
@@ -1,152 +1,78 @@
-from dataclasses import replace
-from typing import Sequence, Tuple
+from typing import Sequence
 
-from common.agent_configuration import (
-    AgentConfiguration,
-    ExploitationConfiguration,
-    ExploitationOptionsConfiguration,
-    NetworkScanConfiguration,
-    PluginConfiguration,
-    PropagationConfiguration,
-    ScanTargetConfiguration,
-)
-from common.credentials import Credentials
-
-from . import TestConfiguration
+from common.agent_configuration import AgentConfiguration, PluginConfiguration
 
 
 def add_exploiters(
     agent_configuration: AgentConfiguration,
     brute_force: Sequence[PluginConfiguration] = [],
     vulnerability: Sequence[PluginConfiguration] = [],
 ) -> AgentConfiguration:
-    exploitation_configuration = replace(
-        agent_configuration.propagation.exploitation,
-        brute_force=brute_force,
-        vulnerability=vulnerability,
-    )
-    return replace_exploitation_configuration(agent_configuration, exploitation_configuration)
+
+    agent_configuration.propagation.exploitation.brute_force = brute_force
+    agent_configuration.propagation.exploitation.vulnerability = vulnerability
+
+    return agent_configuration
```

Comment:
I don't think we need any of these functions anymore. We can just manipulate the configurations directly in the config files. IMO, these no longer add clarity or simplicity.

Reply:
All the places where the configuration needs to modified use these functions. I think that's better than specifying the whole thing in multiple places which is repetitive. Plus, if we ever modify the configuration structure, we'll only have to change this in one place.

Reply:
If you want to leave them, then they should either make a copy and return the modified copy, or not return anything and modify the input.

Reply:
Updated the code to make a copy and return the modified copy.